src/incidents/incidents.service.ts

```diff
@@ -139,6 +168,34 @@ export class IncidentsService {
     });
   }
 
+  async findIncidentTime(id: string): Promise<IncidentTime> {
+    const { events } = await this.db.incident.findUnique({
+      where: { id },
+      include: {
+        events: true,
+      },
+    });
+
+    let causeEvent, detectionEvent, resolutionEvent;
+
+    events.forEach((event) => {
+      switch (event.type) {
+        case EventType.CAUSE:
+          causeEvent = event;
+          break;
+        case EventType.RESOLUTION:
+          resolutionEvent = event;
+          break;
+        case EventType.DETECTION:
+          detectionEvent = event;
+          break;
+      }
+    });
```

Comment:
I think we can clean this up quite a bit by using the `find` method!

```suggestion
    const causeEvent = events.find((event) => event.type === EventType.CAUSE);
    const detection = events.find((event) => event.type === EventType.DETECTION);
    const resolutionEvent = events.find((event) => event.type === EventType.RESOLUTION);
```